index.js

```diff
@@ -54,7 +56,8 @@ function BigList(items, opts) {
     render();
 
     function render() {
-        var scrollPercentage = root.scrollTop / root.scrollHeight;
+        scrollHeight = scrollHeight || root.scrollHeight;
```

Comment:
Does this break on window resizes?

Reply:
I don't think so as scrollHeight is independent of this. From [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollHeight):
> The scrollHeight value is equal to the minimum height the element would require in order to fit all the content in the viewport without using a vertical scrollbar.

Reply:
Does the real item height depend on the contents of the items and the width of the item? It's been a while since I looked at this, but I think the item heights could be independent (and might change as you scroll through items) and that they might increase/decrease in size as you scale the window.

Reply:
No, an individual list item's height is a property enforced by BigList, therefore it won't change. This is required to accurately calculate the height of the 'before' and 'after' padding divs to simulate the entirety of the list.
An item's width can change - therefore potentially causing the browser to render the horizontal scrollbar as the user scrolls vertically. However, this doesn't affect the scrollHeight property.

Reply:
I don't think the height of the list item is enforced. If it were, the scrollHeight could probably be replaced by a simple calculation. The height we provide informs how the list scrolls and the number of items to render. It would mean that as long as the height provided is the minimum of the items displayed, you'll be able to scroll through a list of items of varying sizes. There are some caveats, i.e. the scrolling is linear across the virtual item space, but there is some tolerance for slightly larger items in the list. I want to confirm all of this as this was written 4 years ago, but I do recall checking that this worked with varying sizes of item in the list. 

To test I would consider adding a large item to the end of a list and seeing if we can scroll all the way to the bottom. The thought here is that that scrollHeight is larger when the last item is displayed, but we would have cached a smaller number, meaning that the last item leaks off the end of the list and isn't fully visible when scrolled 100%.

That said, there are probably some things we can do, like cache invalidate when items are switched out. It doesn't prevent overflows when the window is resized, but it will correct itself if scrolled (and window resize behavior is mostly edge case).